vcpkg/users/versioning-troubleshooting.md

```diff
@@ -188,6 +188,57 @@ Resolutions:
       run: git fetch --unshallow
     ```
 
+## <a name="unexpected-default-features"></a> Cause: Unexpected installation of default features in transitive dependencies
+
+When using `vcpkg` to manage your project's dependencies, you might encounter a situation where a transitive dependency installs with its default features, even when you've specified `"depend-defaults": false` in your top-level manifest. This can lead to confusion when features that were intended to be excluded, such as `foo`, are still being installed.
+
+### Scenario
+
+You have a dependency on library `Y`, which in turn depends on library `X`. Library `X` has default features, including `foo`, that you want to exclude from your project. Your top-level manifest for library `Y` might look something like this:
+
+```json
+{
+  "name": "my-project",
+  "dependencies": [
+    {
+      "name": "Y",
+      "features": ["featureB"],
+      "depend-defaults": false
+    }
+  ]
+}
+```
+
+You expect that `X` will be installed without its default features due to the `"depend-defaults": false` setting. However, `X` is still installed with the default feature `foo`.
+
+### Reason
+
+The `depend-defaults` property is only considered at the top-level manifest. This means that default features of transitive dependencies (like `X` in this scenario) are still included unless explicitly disabled at the top level. When library `Y` is resolved, `vcpkg` does not propagate the `"depend-defaults": false` setting to the transitive dependency `X`, resulting in the installation of `X` with its default features.
+
+### Resolution
+
+To ensure that transitive dependencies like `X` are installed without their default features, you need to promote them to direct dependencies in your top-level manifest and explicitly disable their default features. Modify your `vcpkg.json` to include `X` directly with `"default-features": false`:
```

Comment:
This instruction doesn't lead to the desired resolution when the user needs third port `Z` which depends on `X` but lacks the `"default-features": false` property on the `X` dependency just because `Z`'s maintainers weren't aware of the default features (at that time).
- The user can't override the behavior.
- The user will face difficulties to realize that this is a port bug in `Z`.
- The `Z` maintainers have no tool support to check which dependencies have default features
- The `Z` maintainers have no way to express that they really want to depend on default-features.

All up to brainware.